skyportal/handlers/api/plan.py

```diff
@@ -0,0 +1,478 @@
+import datetime
+import copy
+
+from sqlalchemy.orm.exc import NoResultFound
+from sqlalchemy.orm import joinedload
+from astropy import time
+import ephem
+import gwemopt
+import numpy as np
+
+from baselayer.app.access import auth_or_token
+from ..base import BaseHandler
+from ...models import (
+    DBSession,
+    Instrument,
+    Localization,
+    ObservingPlan,
+    PlannedObservation,
+    Telescope,
+)
+
+plan_args_default = {
+    'filt': ['g', 'r', 'g'],
+    'exposuretimes': [300.0, 300.0, 300.0],
+    'doReferences': False,
+    'doUsePrimary': True,
+    'doBalanceExposure': False,
+    'doDither': False,
+    'usePrevious': False,
+    'doCompletedObservations': False,
+    'doPlannedObservations': False,
+    'cobs': [None, None],
+    'schedule_type': 'greedy',
+    'filterScheduleType': 'block',
+    'airmass': 2.5,
+    'schedule_strategy': 'tiling',
+    'mindiff': 30.0 * 60.0,
+    'doMaxTiles': False,
+    'max_nb_tiles': 1000,
+    'doRASlice': True,
+    'raslice': [0.0, 24.0],
+}
+
+
+class PlanHandler(BaseHandler):
+    @auth_or_token
+    def post(self):
+        """
+        ---
+        description: Create an observation plan
+        tags:
+          - observingplans
+        requestBody:
+          content:
+            application/json:
+              schema: PlanHandlerPost
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
```

Comment:
This is incomplete.

Reply:
Because error isn't supposed to be a catch all?

Reply:
The Success object needs to be augmented with any additional return fields.

Reply:
Got it! ok added.